lib/widgets/totals_table.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:flutter/material.dart';
+import '../models/cow_requirements_model.dart';
+
+class TotalsTable extends StatelessWidget {
+  final List<Map<String, dynamic>> fodderItems;
+  final List<Map<String, dynamic>> concentrateItems;
+  final CowRequirements cowRequirements;
+
+  TotalsTable({
+    super.key,
+    required this.fodderItems,
+    required this.concentrateItems,
+    required this.cowRequirements,
+  });
+
+  final List<Map<String, dynamic>> columns = const [
+    {
+      'label': 'DM Intake\n(kg/d)',
+      'key': 'dmIntake',
+      'decimals': 5,
+      'reqKey': 'dmIntake'
+    },
+    {
+      'label': 'ME Intake\n(MJ/d)',
+      'key': 'meIntake',
+      'decimals': 5,
+      'reqKey': 'meIntake'
+    },
+    {
+      'label': 'CP Intake\n(kg/d)',
+      'key': 'cpIntake',
+      'decimals': 5,
+      'reqKey': 'cpIntake'
+    },
+    {
+      'label': 'Ca Intake\n(kg/d)',
+      'key': 'caIntake',
+      'decimals': 5,
+      'reqKey': 'caIntake'
+    },
+    {
+      'label': 'P Intake\n(kg/d)',
+      'key': 'pIntake',
+      'decimals': 5,
+      'reqKey': 'pIntake'
+    },
+    {'label': 'Cost\n(ERN)', 'key': 'cost', 'decimals': 2, 'reqKey': null},
+  ];
+
+  Map<String, double> _calculateTotals() {
+    Map<String, double> totals = {};
+    for (var col in columns) {
+      String key = col['key'] as String;
+      totals[key] = 0;
+    }
+
+    for (var item in [...fodderItems, ...concentrateItems]) {
+      for (var col in columns) {
+        String key = col['key'] as String;
+        totals[key] = (totals[key] ?? 0) + (item[key] as num);
+      }
+    }
+
+    return totals;
+  }
+
+  Widget _buildComparisonIcon(double total, double requirement) {
+    const double threshold = 0.05; // 5% threshold for considering values equal
```

Comment:
I like the consistency of always requiring the answer to be within 5% of the required... However the different aspects have different required accuracies. (Eg. one of the percentages is ±2%, so it's additive, rather than multiplicative because it's already a percentage). Please review the google sheet, and update accordingly.